### PR TITLE
Cleanup

### PR DESCRIPTION
**Pull request by kevmoo**

### CODE REVIEW

File: lib/src/change_notifier.dart [hunk 1]
```diff
@@ -60,13 +60,13 @@ class ChangeNotifier<C extends ChangeRecord> implements Observable<C> {
...
   bool deliverChanges() {
-    List<ChangeRecord> changes;
+    List<C> changes;
     if (_scheduled && hasObservers) {
       if (_queue != null) {
         changes = freezeInDevMode(_queue);
         _queue = null;
       } else {
-        changes = ChangeRecord.ANY;
+        changes = <C>[];
```
Comment (jmesserly) [inline lib/src/change_notifier.dart]: fyi, `<C>` is not needed, inference will do this for you given `changes = [];` ... nothing wrong with being explicit tho

File: lib/src/differs/list_differ.dart [hunk 2]
```diff
@@ -206,7 +206,7 @@ List<ListChangeRecord/*<E>*/ > _calcSplices/*<E>*/(
   oldEnd -= suffixCount;
 
   if (currentEnd - currentStart == 0 && oldEnd - oldStart == 0) {
```
Comment (jmesserly) [inline lib/src/differs/list_differ.dart:208]: fix the _calcSplices to remove the generic method comments and use generic method syntax?

File: lib/src/differs/list_differ.dart [hunk 3]
```diff
@@ -206,7 +206,7 @@ List<ListChangeRecord/*<E>*/ > _calcSplices/*<E>*/(
...
-    return ChangeRecord.NONE;
+    return <ListChangeRecord<E>>[];
```
Comment (jmesserly) [inline lib/src/differs/list_differ.dart]: this could also be `[]` if you want

File: lib/src/differs/map_differ.dart [hunk 4]
```diff
@@ -17,7 +17,7 @@ class MapDiffer<K, V> implements Differ<Map<K, V>> {
   @override
   List<MapChangeRecord<K, V>> diff(Map<K, V> oldValue, Map<K, V> newValue) {
     if (identical(oldValue, newValue)) {
-      return ChangeRecord.NONE;
+      return <MapChangeRecord<K, V>>[];
```
Comment (jmesserly) [inline lib/src/differs/map_differ.dart]: could be `[]`

File: lib/src/differs/set_differ.dart [hunk 5]
```diff
@@ -17,7 +17,7 @@ class SetDiffer<E> implements Differ<Set<E>> {
   @override
   List<SetChangeRecord<E>> diff(Set<E> oldValue, Set<E> newValue) {
     if (identical(oldValue, newValue)) {
-      return ChangeRecord.NONE;
+      return <SetChangeRecord<E>>[];
```
Comment (jmesserly) [inline lib/src/differs/set_differ.dart]: could be `[]`

File: lib/src/internal.dart [hunk 6]
```diff
@@ -2,10 +2,10 @@
 // for details. All rights reserved. Use of this source code is governed by a
 // BSD-style license that can be found in the LICENSE file.
 
-List/*<E>*/ freezeInDevMode/*<E>*/(List/*<E>*/ list) {
+List<E> freezeInDevMode<E>(List<E> list) {
   if (list == null) return const [];
   assert(() {
-    list = new List/*<E>*/ .unmodifiable(list);
+    list = new List<E>.unmodifiable(list);
```
Comment (jmesserly) [inline lib/src/internal.dart]: could be `new List.unmodifiable`

**Review by jmesserly — APPROVED** [review]
a few optional suggestions but LGTM

**Comment by kevmoo** [comment]
@jmesserly  – thanks! did all of those...

File: lib/src/differs/list_differ.dart (hunk 2 above)
Comment (kevmoo) [reply lib/src/differs/list_differ.dart:208]: Done!

**Comment by kevmoo** [comment]
@matanlurey ? LGTY?

**Comment by natebosch** [comment]
Will this get landed?

**Comment by kevmoo** [comment]
> Will this get landed?

I wanted @matanlurey to take a look...but it doesn't affect angular at all.

This does affect components, though. Uh...so maybe @nshahan or @TedSander should look?

**Comment by nshahan** [comment]
Are there plans to land this internally?

**Comment by kevmoo** [comment]
🤷‍♂️ 

I was using it and running into issues.

I know components uses observable – so it seems like it's something we should maintain, right?

**Pull request closed without merging**